dbms/src/Storages/Transaction/RegionData.cpp

```diff
@@ -310,7 +324,13 @@ void RegionData::OrphanKeysInfo::observeExtraKey(TiKVKey && key)
 
 bool RegionData::OrphanKeysInfo::observeKeyFromNormalWrite(const TiKVKey & key)
 {
-    return remained_keys.erase(key);
+    bool res = remained_keys.erase(key);
+    if (res)
+    {
+        // If we erased something, log that.
+        removed_remained_keys.insert(TiKVKey::copyFromObj(key));
```

Comment:
When will we remove the keys stored in `removed_remained_keys`?

Reply:
This is for debug. It detects the following:
1. Observe a write key in snapshot and put it into orphan info
2. Observe this write key in raft log
3. Observe this write key again in raft log

Will this actually happen? We don't know. However, we can trace some information here with limited cost(some memory cost).

However, since the check is temporarily disabled. I think I can comment this for now.

Reply:
I think we'd better comment out this line now. Or it is easy for us to forget the "removed keys" will be stored in memory for a long time